src/main/java/core/basesyntax/service/impl/CsvReportGenerator.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class CsvReportGenerator implements ReportGenerator {
+    private static final String REPORT_FIRST_LINE = "fruit,quantity";
+    private static final String COMA = ",";
+
+    @Override
+    public String generateReport(Map<String, Integer> storageMap) {
+        StringBuilder stringBuilder = new StringBuilder(REPORT_FIRST_LINE + "\n");
+        for (Entry<String, Integer> entry : storageMap.entrySet()) {
+            stringBuilder.append(entry.getKey()).append(COMA).append(entry.getValue());
+            stringBuilder.append(System.lineSeparator());
+        }
+        return stringBuilder.toString().trim();

```

Comment:
```suggestion
        return stringBuilder.toString();

```

Reply:
this `trim()` can be removed by placing the new line **before** appending the next data line, and removing `\n` from the `StringBuilder` constructor

Reply:
Not resolved

Reply:
Not resolved

Reply:
my fault)

Reply:
Still not resolved